src/main/java/com/rummgp/medical_clinic/repository/PatientRepository.java

```diff
@@ -0,0 +1,56 @@
+package com.rummgp.medical_clinic.repository;
+
+import com.rummgp.medical_clinic.model.Patient;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class PatientRepository {
+    private List<Patient> patients = new ArrayList<>();
+
+    public List<Patient> findAll() {
+        return new ArrayList<>(patients);
+    }
+
+    public Optional<Patient> findByEmail(String email) {
+        return patients.stream()
+                .filter(patient -> patient.getEmail().equals(email))
+                .findFirst();
+    }
+
+    public boolean add(Patient patient) {
+        if (findByEmail(patient.getEmail()).isPresent()) {
+            return false;
+        }
+        patients.add(patient);
+        return true;
+    }
+
+    public boolean remove(String email) {
+        return patients.removeIf(patient -> patient.getEmail().equals(email));
+    }
+
+    public boolean edit(String email, Patient updated) {
```

Comment:
To nie jest edycja tylko to jest nadpisanie (edit != delete + add) 

existingPatient = findByEmail(email).ifPresent{ patient ->
patient.setEmail(updatedPatient.getEmail());
patient.setName(updatedPatient.getName());
/// pozostale pola
}

patient.setEmail(updatedPatient.getEmail());
patient.setName(updatedPatient.getName());
/// pozostale pola
 te linijki powyzej to ustawienie nowych wartosci pol 
 mozemy w klasie patient dodać metode w Patient:edit(Patient newData) {
 this.email = newData.getEmail()
 this.name = newData.getName();
 }